order/views.py

```diff
@@ -1,27 +1,37 @@
 from rest_framework.response import Response
-from rest_framework import authentication
 from rest_framework.views import APIView
+from rest_framework import authentication
 from rest_framework import status
 
+from django.shortcuts import get_object_or_404
+
 from order.serializer import OrderSerializer, OrderdetailsSerializer
 from order.models import Order, OrderDetails
 from user.utils import admin_required
-# Create your views here.
 
+# Create your views here.
 
 class OrderView(APIView):
     serializer_class = OrderSerializer
     permission_classes = [admin_required]
     authentication_classes = [authentication.TokenAuthentication]
     
     def get(self, request, id=None):
-        order = Order.objects.all()
-        start_date = request.GET.get('start_date', None)
-        end_date = request.GET.get('end_date', None)
-        if start_date is not None or end_date is not None :
-            order = order.filter(date_time =[start_date, end_date]) 
-        serializer = self.serializer_class(order, many=True)
-        return Response(serializer.data)
+        city = request.GET.get('city')
+        order = Order.objects.filter(user__user_profile__city__icontains=city)
```

Comment:
What if city field is None? Need to fix this. If nothing is there for filtering then all orders should be returned.

Also add pagination for this also.